### DISCUSSION TITLE
Add unique highlighting group to current viewing buffer/file

### DISCUSSION THREAD

**Discussion by ugoa** [nvim-tree/nvim-tree.lua#2925] [Q&A] [answered]
**Can this functionality be implemented utilising API?**

I'd say no? I have searched the codebase and options but got no clue of how to support this with existing APIs.

**Is your feature request related to a problem? Please describe.**

The current option in nvimtree to highlight opened files is controlled by option `renderer.highlight_opened_files`, which would highlight all currently opened files with one highlight group [NvimTreeOpenedHL](https://github.com/nvim-tree/nvim-tree.lua/blob/8405ecfbd6bb08a94ffc9c68fef211eea56e8a3b/lua/nvim-tree/appearance/init.lua#L89). This causes all the files share the same color scheme, and it becomes quite difficult to identify the current viewing buffer in the file tree.

I wonder if an option can be added to make the current viewing buffer to use a different hightlight group? something like `highlight_current_buffer`. I imagine it be a combination of option `update_focused_file.enabled` and `renderer.highlight_opened_files`. 

Something like the below effect:

![ss](https://github.com/user-attachments/assets/0ae0a184-3a28-45a8-9687-42a868a065db)

**Describe the solution you'd like**
Add new option toggle this feature, such as `highlight_current_buffer`

**Describe alternatives you've considered**
Maybe add new value `single` to existing option `renderer.highlight_opened_files`?

**Additional context**
Please see the screenshot

**Comment by alex-courtis** [comment, marked as answer]
It's possible to add those highlight groups, a [pull request](https://github.com/nvim-tree/nvim-tree.lua/blob/master/CONTRIBUTING.md) would be most gratefully appreciated.

However, I don't think it is really necessary: the opened file is indicated by the cursorline, which overrides the opened files highlight.

It looks like you've got `&cursorline` and `&number` disabled. Turn one or both on and then change their highlight:

setup:
```lua
  ---
  view = {
    number = true,
    cursorline = true,
    ---
```

highlight:
```vim
:hi NvimTreeCursorLine guifg=blue
:hi NvimTreeCursorLineNr guifg=red
```

**Comment by alex-courtis** [comment]
If you're happy with the above I'll move this to [Q&A Discussions](https://github.com/nvim-tree/nvim-tree.lua/discussions/categories/q-a)

**Comment by ugoa** [comment]
Thank you so much, this is beautiful and exactly what I wanted. Really appreciate it!
